repos/system_upgrade/el8toel9/actors/fixgrubefiwrapper/actor.py

```diff
@@ -0,0 +1,34 @@
+import os
+
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.stdlib import api, run
+from leapp.models import FirmwareFacts, TransactionCompleted
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, RPMUpgradePhaseTag
+
+
+class FixGrubEFIWrapper(Actor):
+    """
+    Update /boot/efi/EFI/redhat/grub.cfg on UEFI systems.
+
+    See [RHEL-36186](https://issues.redhat.com/browse/RHEL-36186).
+    """
+
+    name = 'fix_grub_efi_wrapper'
+    consumes = (TransactionCompleted, FirmwareFacts)
+    produces = (Report,)
+    tags = (RPMUpgradePhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        ff = next(self.consume(FirmwareFacts), None)
+        if not ff:
+            raise StopActorExecution(
+                'Could not identify system firmware',
+                details={'details': 'Actor did not receive FirmwareFacts message.'}
+            )
+
+        if ff.firmware == 'efi' and os.path.ismount('/boot/efi') and os.path.isfile('/boot/efi/EFI/redhat/grub.cfg'):
+            api.current_logger().info('Removing --root-dev-only if present.')
+            run(['/usr/bin/sed', '-i', r's/--root-dev-only\s*//', '/boot/efi/EFI/redhat/grub.cfg'])
```

Comment:
I prefer to be careful and catch exceptions, to be sure it doesn't crash here (I do not expect non-zero exit code here at all, but who knows :shrug: ):
```suggestion
            try:
                run(['/usr/bin/sed', '-i', r's/--root-dev-only\s*//', '/boot/efi/EFI/redhat/grub.cfg'])
            except (OSError, CalledProcessError):
                api.current_logger().warning('Cannot apply the fix of /boot/efi/EFI/redhat/grub.cfg.')
                return
```